global: migrate to quarkus version 3.10

rekt-hard opened the pull request:
<!-- Please make sure you have read the submission guidelines before posting an PR -->
<!-- https://github.com/tuwien-csd/damap-backend/blob/next/CONTRIBUTING.md#pullrequests -->

## Description
<!-- Please select a type that best describes your PR -->
Migration to the latest quarkus version.

Most changes are only about moving from `javax` to `jakarta` and deprecation warning updates.
With the move to the new quarkus version, the hibernate ORM version will also change from `v5.6.` to `v6.2`. A db change is required, which will add sequences - one for each tables id.

After the upgrade, additional output about running SQL update commands is printed:
<details><summary>Additional output</summary>
<p>


```sql
The following SQL may resolve the database issues, as generated by the Hibernate schema migration tool. WARNING: You must manually verify this SQL is correct, this is a best effort guess, do not copy/paste it without verifying that it does what you expect.

alter table if exists access_management alter column id set data type bigint;
alter table if exists access_management alter column start_date set data type timestamp(6);
alter table if exists access_management alter column until_date set data type timestamp(6);
alter table if exists access_management alter column version set data type bigint;
alter table if exists access_management alter column dmp_id set data type bigint;
alter table if exists access_management alter column identifier_id set data type bigint;
alter table if exists access_management_AUD alter column id set data type bigint;
alter table if exists access_management_AUD alter column REV set data type bigint;
alter table if exists access_management_AUD alter column REVTYPE set data type smallint;
alter table if exists access_management_AUD alter column start_date set data type timestamp(6);
alter table if exists access_management_AUD alter column until_date set data type timestamp(6);
alter table if exists access_management_AUD alter column dmp_id set data type bigint;
alter table if exists access_management_AUD alter column identifier_id set data type bigint;
alter table if exists Administration alter column id set data type bigint;
alter table if exists Administration alter column start_date set data type timestamp(6);
alter table if exists Administration alter column until_date set data type timestamp(6);
alter table if exists Administration alter column version set data type bigint;
alter table if exists Administration_AUD alter column id set data type bigint;
alter table if exists Administration_AUD alter column REV set data type bigint;
alter table if exists Administration_AUD alter column REVTYPE set data type smallint;
alter table if exists Administration_AUD alter column start_date set data type timestamp(6);
alter table if exists Administration_AUD alter column until_date set data type timestamp(6);
alter table if exists Consent alter column id set data type bigint;
alter table if exists Consent alter column given_date set data type timestamp(6);
alter table if exists Contributor alter column id set data type bigint;
alter table if exists Contributor alter column version set data type bigint;
alter table if exists Contributor alter column affiliation_id set data type bigint;
alter table if exists Contributor alter column dmp_id set data type bigint;
alter table if exists Contributor alter column person_id set data type bigint;
alter table if exists Contributor_AUD alter column id set data type bigint;
alter table if exists Contributor_AUD alter column REV set data type bigint;
alter table if exists Contributor_AUD alter column REVTYPE set data type smallint;
alter table if exists Contributor_AUD alter column affiliation_id set data type bigint;
alter table if exists Contributor_AUD alter column dmp_id set data type bigint;
alter table if exists Contributor_AUD alter column person_id set data type bigint;
alter table if exists Cost alter column id set data type bigint;
alter table if exists Cost alter column description set data type varchar(255);
alter table if exists Cost alter column cost_value set data type float4;
alter table if exists Cost alter column version set data type bigint;
alter table if exists Cost alter column dmp_id set data type bigint;
alter table if exists Cost_AUD alter column id set data type bigint;
alter table if exists Cost_AUD alter column REV set data type bigint;
alter table if exists Cost_AUD alter column REVTYPE set data type smallint;
alter table if exists Cost_AUD alter column description set data type varchar(255);
alter table if exists Cost_AUD alter column cost_value set data type float4;
alter table if exists Cost_AUD alter column dmp_id set data type bigint;
alter table if exists data_quality alter column Dmp_id set data type bigint;
alter table if exists data_quality_AUD alter column REV set data type bigint;
alter table if exists data_quality_AUD alter column Dmp_id set data type bigint;
alter table if exists data_quality_AUD alter column REVTYPE set data type smallint;
alter table if exists data_type alter column Dataset_id set data type bigint;
alter table if exists data_type_AUD alter column REV set data type bigint;
alter table if exists data_type_AUD alter column Dataset_id set data type bigint;
alter table if exists data_type_AUD alter column REVTYPE set data type smallint;
alter table if exists Dataset alter column id set data type bigint;
alter table if exists Dataset alter column date_of_deletion set data type timestamp(6);
alter table if exists Dataset alter column description set data type oid;
alter table if exists Dataset alter column reason_for_deletion set data type varchar(255);
alter table if exists Dataset alter column data_size set data type bigint;
alter table if exists Dataset alter column start_date set data type timestamp(6);
alter table if exists Dataset alter column version set data type bigint;
alter table if exists Dataset alter column dataset_pid set data type bigint;
alter table if exists Dataset alter column deletion_person_id set data type bigint;
alter table if exists Dataset alter column dmp_id set data type bigint;
alter table if exists Dataset_AUD alter column id set data type bigint;
alter table if exists Dataset_AUD alter column REV set data type bigint;
alter table if exists Dataset_AUD alter column REVTYPE set data type smallint;
alter table if exists Dataset_AUD alter column date_of_deletion set data type timestamp(6);
alter table if exists Dataset_AUD alter column description set data type varchar(255);
alter table if exists Dataset_AUD alter column reason_for_deletion set data type varchar(255);
alter table if exists Dataset_AUD alter column data_size set data type bigint;
alter table if exists Dataset_AUD alter column start_date set data type timestamp(6);
alter table if exists Dataset_AUD alter column dataset_pid set data type bigint;
alter table if exists Dataset_AUD alter column deletion_person_id set data type bigint;
alter table if exists Dataset_AUD alter column dmp_id set data type bigint;
alter table if exists Distribution alter column id set data type bigint;
alter table if exists Distribution alter column version set data type bigint;
alter table if exists Distribution alter column dataset_id set data type bigint;
alter table if exists Distribution alter column host_id set data type bigint;
alter table if exists Distribution_AUD alter column id set data type bigint;
alter table if exists Distribution_AUD alter column REV set data type bigint;
alter table if exists Distribution_AUD alter column REVTYPE set data type smallint;
alter table if exists Distribution_AUD alter column dataset_id set data type bigint;
alter table if exists Distribution_AUD alter column host_id set data type bigint;
alter table if exists Dmp alter column id set data type bigint;
alter table if exists Dmp alter column closed_access_info set data type varchar(255);
alter table if exists Dmp alter column created set data type timestamp(6);
alter table if exists Dmp alter column data_generation set data type varchar(255);
alter table if exists Dmp alter column data_rights_access_control set data type varchar(255);
alter table if exists Dmp alter column description set data type CLOB;
alter table if exists Dmp alter column documentation set data type varchar(255);
alter table if exists Dmp alter column external_storage_info set data type varchar(255);
alter table if exists Dmp alter column legal_restrictions_comment set data type varchar(255);
alter table if exists Dmp alter column metadata set data type varchar(255);
alter table if exists Dmp alter column modified set data type timestamp(6);
alter table if exists Dmp alter column no_data_explanation set data type varchar(255);
alter table if exists Dmp alter column other_data_quality set data type varchar(255);
alter table if exists Dmp alter column other_data_sec_measures set data type varchar(255);
alter table if exists Dmp alter column other_legal_r_documents set data type varchar(255);
alter table if exists Dmp alter column other_personal_data_compliance set data type varchar(255);
alter table if exists Dmp alter column restricted_access_info set data type varchar(255);
alter table if exists Dmp alter column restricted_data_access set data type varchar(255);
alter table if exists Dmp alter column sensitive_data_access set data type varchar(255);
alter table if exists Dmp alter column structure set data type varchar(255);
alter table if exists Dmp alter column target_audience set data type varchar(255);
alter table if exists Dmp alter column tools set data type varchar(255);
alter table if exists Dmp alter column version set data type bigint;
alter table if exists Dmp alter column project_id set data type bigint;
alter table if exists Dmp_AUD alter column id set data type bigint;
alter table if exists Dmp_AUD alter column REV set data type bigint;
alter table if exists Dmp_AUD alter column REVTYPE set data type smallint;
alter table if exists Dmp_AUD alter column closed_access_info set data type varchar(255);
alter table if exists Dmp_AUD alter column created set data type timestamp(6);
alter table if exists Dmp_AUD alter column data_generation set data type varchar(255);
alter table if exists Dmp_AUD alter column data_rights_access_control set data type varchar(255);
alter table if exists Dmp_AUD alter column description set data type CLOB;
alter table if exists Dmp_AUD alter column documentation set data type varchar(255);
alter table if exists Dmp_AUD alter column external_storage_info set data type varchar(255);
alter table if exists Dmp_AUD alter column legal_restrictions_comment set data type varchar(255);
alter table if exists Dmp_AUD alter column metadata set data type varchar(255);
alter table if exists Dmp_AUD alter column modified set data type timestamp(6);
alter table if exists Dmp_AUD alter column no_data_explanation set data type varchar(255);
alter table if exists Dmp_AUD alter column other_data_quality set data type varchar(255);
alter table if exists Dmp_AUD alter column other_data_sec_measures set data type varchar(255);
alter table if exists Dmp_AUD alter column other_legal_r_documents set data type varchar(255);
alter table if exists Dmp_AUD alter column other_personal_data_compliance set data type varchar(255);
alter table if exists Dmp_AUD alter column restricted_access_info set data type varchar(255);
alter table if exists Dmp_AUD alter column restricted_data_access set data type varchar(255);
alter table if exists Dmp_AUD alter column sensitive_data_access set data type varchar(255);
alter table if exists Dmp_AUD alter column structure set data type varchar(255);
alter table if exists Dmp_AUD alter column target_audience set data type varchar(255);
alter table if exists Dmp_AUD alter column tools set data type varchar(255);
alter table if exists Dmp_AUD alter column project_id set data type bigint;
alter table if exists dmp_version alter column id set data type bigint;
alter table if exists dmp_version alter column revision_number set data type bigint;
alter table if exists dmp_version alter column version_date set data type timestamp(6);
alter table if exists dmp_version alter column dmp_id set data type bigint;
alter table if exists external_storage alter column id set data type bigint;
alter table if exists external_storage_AUD alter column id set data type bigint;
alter table if exists external_storage_AUD alter column REV set data type bigint;
alter table if exists Funding alter column id set data type bigint;
alter table if exists Funding alter column version set data type bigint;
alter table if exists Funding alter column funder_id set data type bigint;
alter table if exists Funding alter column grant_id set data type bigint;
alter table if exists Funding_AUD alter column id set data type bigint;
alter table if exists Funding_AUD alter column REV set data type bigint;
alter table if exists Funding_AUD alter column REVTYPE set data type smallint;
alter table if exists Funding_AUD alter column funder_id set data type bigint;
alter table if exists Funding_AUD alter column grant_id set data type bigint;
alter table if exists Host alter column discriminator set data type varchar(31);
alter table if exists Host alter column id set data type bigint;
alter table if exists Host alter column version set data type bigint;
alter table if exists Host alter column dmp_id set data type bigint;
alter table if exists Host_AUD alter column id set data type bigint;
alter table if exists Host_AUD alter column REV set data type bigint;
alter table if exists Host_AUD alter column discriminator set data type varchar(31);
alter table if exists Host_AUD alter column REVTYPE set data type smallint;
alter table if exists Host_AUD alter column dmp_id set data type bigint;
alter table if exists Identifier alter column id set data type bigint;
alter table if exists Identifier alter column version set data type bigint;
alter table if exists Identifier_AUD alter column id set data type bigint;
alter table if exists Identifier_AUD alter column REV set data type bigint;
alter table if exists Identifier_AUD alter column REVTYPE set data type smallint;
alter table if exists inter_storage_translation alter column id set data type bigint;
alter table if exists inter_storage_translation alter column description set data type varchar(255);
alter table if exists inter_storage_translation alter column version set data type bigint;
alter table if exists inter_storage_translation alter column internal_storage_id set data type bigint;
alter table if exists inter_storage_translation_AUD alter column id set data type bigint;
alter table if exists inter_storage_translation_AUD alter column REV set data type bigint;
alter table if exists inter_storage_translation_AUD alter column REVTYPE set data type smallint;
alter table if exists inter_storage_translation_AUD alter column description set data type varchar(255);
alter table if exists inter_storage_translation_AUD alter column internal_storage_id set data type bigint;
alter table if exists internal_storage alter column id set data type bigint;
alter table if exists internal_storage alter column version set data type bigint;
alter table if exists internal_storage_AUD alter column id set data type bigint;
alter table if exists internal_storage_AUD alter column REV set data type bigint;
alter table if exists internal_storage_AUD alter column REVTYPE set data type smallint;
alter table if exists legal_restr_documents alter column Dmp_id set data type bigint;
alter table if exists legal_restr_documents_AUD alter column REV set data type bigint;
alter table if exists legal_restr_documents_AUD alter column Dmp_id set data type bigint;
alter table if exists legal_restr_documents_AUD alter column REVTYPE set data type smallint;
alter table if exists personal_data_compliance alter column Dmp_id set data type bigint;
alter table if exists personal_data_compliance_AUD alter column REV set data type bigint;
alter table if exists personal_data_compliance_AUD alter column Dmp_id set data type bigint;
alter table if exists personal_data_compliance_AUD alter column REVTYPE set data type smallint;
alter table if exists Project alter column id set data type bigint;
alter table if exists Project alter column description set data type CLOB;
alter table if exists Project alter column project_end set data type timestamp(6);
alter table if exists Project alter column project_start set data type timestamp(6);
alter table if exists Project alter column version set data type bigint;
alter table if exists Project alter column funding_id set data type bigint;
alter table if exists Project_AUD alter column id set data type bigint;
alter table if exists Project_AUD alter column REV set data type bigint;
alter table if exists Project_AUD alter column REVTYPE set data type smallint;
alter table if exists Project_AUD alter column description set data type CLOB;
alter table if exists Project_AUD alter column project_end set data type timestamp(6);
alter table if exists Project_AUD alter column project_start set data type timestamp(6);
alter table if exists Project_AUD alter column funding_id set data type bigint;
alter table if exists Repository alter column id set data type bigint;
alter table if exists Repository_AUD alter column id set data type bigint;
alter table if exists Repository_AUD alter column REV set data type bigint;
alter table if exists REVCHANGES alter column REV set data type bigint;
alter table if exists revinfo alter column id set data type bigint;
alter table if exists revinfo alter column timestamp set data type timestamp(6);
alter table if exists sensitive_data_security alter column Dmp_id set data type bigint;
alter table if exists sensitive_data_security_AUD alter column REV set data type bigint;
alter table if exists sensitive_data_security_AUD alter column Dmp_id set data type bigint;
alter table if exists sensitive_data_security_AUD alter column REVTYPE set data type smallint;
alter table if exists Storage alter column id set data type bigint;
alter table if exists Storage alter column internal_storage_id set data type bigint;
alter table if exists Storage_AUD alter column id set data type bigint;
alter table if exists Storage_AUD alter column REV set data type bigint;
alter table if exists Storage_AUD alter column internal_storage_id set data type bigint;
alter table if exists external_storage_AUD add constraint FK7b7c90y7fcli84bku60d49p4w foreign key (REV, id) references Host_AUD;
alter table if exists Repository_AUD add constraint FKds19jc1ncekq21qjuphvryhwn foreign key (REV, id) references Host_AUD;
alter table if exists Storage_AUD add constraint FK7yoclxrm7oinjbbqukpgi4vge foreign key (REV, id) references Host_AUD;
```
</p>
</details>


I have also created a change set for this, to use liquibase specific commands:
<details><summary>Liquibase command change set</summary>
<p>

```yaml
  - changeSet:
      id: 9
      author: David Eckhard
      changes:
      # access_management
       - modifyDataType:
          columnName: id
          newDataType: bigint
          tableName: access_management
       - modifyDataType:
          columnName: version
          newDataType: bigint
          tableName: access_management
       - modifyDataType:
          columnName: dmp_id
          newDataType: bigint
          tableName: access_management
       - modifyDataType:
          columnName: identifier_id
          newDataType: bigint
          tableName: access_management

      # access_management_aud
       - modifyDataType:
          columnName: id
          newDataType: bigint
          tableName: access_management_aud
       - modifyDataType:
          columnName: REV
          newDataType: bigint
          tableName: access_management_aud
       - modifyDataType:
          columnName: REVTYPE
          newDataType: tinyint
          tableName: access_management_aud
       - modifyDataType:
          columnName: dmp_id
          newDataType: bigint
          tableName: access_management_aud
       - modifyDataType:
          columnName: identifier_id
          newDataType: bigint
          tableName: access_management_aud

      # administration
       - modifyDataType:
          columnName: id
          newDataType: bigint
          tableName: administration
       - modifyDataType:
          columnName: version
          newDataType: bigint
          tableName: administration

      # administration_aud
       - modifyDataType:
          columnName: id
          newDataType: bigint
          tableName: administration_aud
       - modifyDataType:
          columnName: REV
          newDataType: bigint
          tableName: administration_aud
       - modifyDataType:
          columnName: REVTYPE
          newDataType: tinyint
          tableName: administration_aud

      # consent
       - modifyDataType:
          columnName: id
          newDataType: bigint
          tableName: consent

      # contributor
       - modifyDataType:
          columnName: id
          newDataType: bigint
          tableName: contributor
       - modifyDataType:
          columnName: version
          newDataType: bigint
          tableName: contributor
       - modifyDataType:
          columnName: affiliation_id
          newDataType: bigint
          tableName: contributor
       - modifyDataType:
          columnName: dmp_id
          newDataType: bigint
          tableName: contributor
       - modifyDataType:
          columnName: person_id
          newDataType: bigint
          tableName: contributor

      # contributor_aud
       - modifyDataType:
          columnName: id
          newDataType: bigint
          tableName: contributor_aud
       - modifyDataType:
          columnName: affiliation_id
          newDataType: bigint
          tableName: contributor_aud
       - modifyDataType:
          columnName: dmp_id
          newDataType: bigint
          tableName: contributor_aud
       - modifyDataType:
          columnName: person_id
          newDataType: bigint
          tableName: contributor_aud
       - modifyDataType:
          columnName: REV
          newDataType: bigint
          tableName: contributor_aud
       - modifyDataType:
          columnName: REVTYPE
          newDataType: tinyint
          tableName: contributor_aud

      # cost
       - modifyDataType:
          columnName: id
          newDataType: bigint
          tableName: cost
       - modifyDataType:
          columnName: cost_value
          newDataType: float
          tableName: cost
       - modifyDataType:
          columnName: version
          newDataType: bigint
          tableName: cost
       - modifyDataType:
          columnName: dmp_id
          newDataType: bigint
          tableName: cost

      # cost_aud
       - modifyDataType:
          columnName: id
          newDataType: bigint
          tableName: cost_aud
       - modifyDataType:
          columnName: REV
          newDataType: bigint
          tableName: cost_aud
       - modifyDataType:
          columnName: REVTYPE
          newDataType: tinyint
          tableName: cost_aud
       - modifyDataType:
          columnName: cost_value
          newDataType: float
          tableName: cost_aud
       - modifyDataType:
          columnName: dmp_id
          newDataType: bigint
          tableName: cost_aud

      # data_quality
       - modifyDataType:
          columnName: dmp_id
          newDataType: bigint
          tableName: data_quality

      # data_quality_aud
       - modifyDataType:
          columnName: dmp_id
          newDataType: bigint
          tableName: data_quality_aud
       - modifyDataType:
          columnName: REV
          newDataType: bigint
          tableName: data_quality_aud
       - modifyDataType:
          columnName: REVTYPE
          newDataType: tinyint
          tableName: data_quality_aud

      # data_type
       - modifyDataType:
          columnName: dataset_id
          newDataType: bigint
          tableName: data_type

      # data_type_aud
       - modifyDataType:
          columnName: dataset_id
          newDataType: bigint
          tableName: data_type_aud
       - modifyDataType:
          columnName: REV
          newDataType: bigint
          tableName: data_type_aud
       - modifyDataType:
          columnName: REVTYPE
          newDataType: tinyint
          tableName: data_type_aud

      # dataset
       - modifyDataType:
          columnName: id
          newDataType: bigint
          tableName: dataset
       - modifyDataType:
          columnName: data_size
          newDataType: bigint
          tableName: dataset
       - modifyDataType:
          columnName: version
          newDataType: bigint
          tableName: dataset
       - modifyDataType:
          columnName: dataset_pid
          newDataType: bigint
          tableName: dataset
       - modifyDataType:
          columnName: deletion_person_id
          newDataType: bigint
          tableName: dataset
       - modifyDataType:
          columnName: dmp_id
          newDataType: bigint
          tableName: dataset

      # dataset_aud
       - modifyDataType:
          columnName: id
          newDataType: bigint
          tableName: dataset_aud
       - modifyDataType:
          columnName: REV
          newDataType: bigint
          tableName: dataset_aud
       - modifyDataType:
          columnName: REVTYPE
          newDataType: tinyint
          tableName: dataset_aud
       - modifyDataType:
          columnName: data_size
          newDataType: bigint
          tableName: dataset_aud
       - modifyDataType:
          columnName: dataset_pid
          newDataType: bigint
          tableName: dataset_aud
       - modifyDataType:
          columnName: deletion_person_id
          newDataType: bigint
          tableName: dataset_aud
       - modifyDataType:
          columnName: dmp_id
          newDataType: bigint
          tableName: dataset_aud

      # distribution
       - modifyDataType:
          columnName: id
          newDataType: bigint
          tableName: distribution
       - modifyDataType:
          columnName: version
          newDataType: bigint
          tableName: distribution
       - modifyDataType:
          columnName: dataset_id
          newDataType: bigint
          tableName: distribution
       - modifyDataType:
          columnName: host_id
          newDataType: bigint
          tableName: distribution

      # distribution_aud
       - modifyDataType:
          columnName: id
          newDataType: bigint
          tableName: distribution_aud
       - modifyDataType:
          columnName: REV
          newDataType: bigint
          tableName: distribution_aud
       - modifyDataType:
          columnName: REVTYPE
          newDataType: tinyint
          tableName: distribution_aud
       - modifyDataType:
          columnName: dataset_id
          newDataType: bigint
          tableName: distribution_aud
       - modifyDataType:
          columnName: host_id
          newDataType: bigint
          tableName: distribution_aud

      # dmp
       - modifyDataType:
          columnName: id
          newDataType: bigint
          tableName: dmp
       - modifyDataType:
          columnName: version
          newDataType: bigint
          tableName: dmp
       - modifyDataType:
          columnName: project_id
          newDataType: bigint
          tableName: dmp

      # dmp_aud
       - modifyDataType:
          columnName: id
          newDataType: bigint
          tableName: dmp_aud
       - modifyDataType:
          columnName: REV
          newDataType: bigint
          tableName: dmp_aud
       - modifyDataType:
          columnName: REVTYPE
          newDataType: tinyint
          tableName: dmp_aud
       - modifyDataType:
          columnName: project_id
          newDataType: bigint
          tableName: dmp_aud

      # dmp_version
       - modifyDataType:
          columnName: id
          newDataType: bigint
          tableName: dmp_version
       - modifyDataType:
          columnName: revision_number
          newDataType: bigint
          tableName: dmp_version
       - modifyDataType:
          columnName: dmp_id
          newDataType: bigint
          tableName: dmp_version

      # external_storage
       - modifyDataType:
          columnName: id
          newDataType: bigint
          tableName: external_storage

      # external_storage_aud
       - modifyDataType:
          columnName: id
          newDataType: bigint
          tableName: external_storage_aud
       - modifyDataType:
          columnName: REV
          newDataType: bigint
          tableName: external_storage_aud

      # funding
       - modifyDataType:
          columnName: id
          newDataType: bigint
          tableName: funding
       - modifyDataType:
          columnName: version
          newDataType: bigint
          tableName: funding
       - modifyDataType:
          columnName: funder_id
          newDataType: bigint
          tableName: funding
       - modifyDataType:
          columnName: grant_id
          newDataType: bigint
          tableName: funding

      # funding_aud
       - modifyDataType:
          columnName: id
          newDataType: bigint
          tableName: funding_aud
       - modifyDataType:
          columnName: REV
          newDataType: bigint
          tableName: funding_aud
       - modifyDataType:
          columnName: REVTYPE
          newDataType: tinyint
          tableName: funding_aud
       - modifyDataType:
          columnName: funder_id
          newDataType: bigint
          tableName: funding_aud
       - modifyDataType:
          columnName: grant_id
          newDataType: bigint
          tableName: funding_aud

      # host
       - modifyDataType:
          columnName: id
          newDataType: bigint
          tableName: host
       - modifyDataType:
          columnName: version
          newDataType: bigint
          tableName: host
       - modifyDataType:
          columnName: dmp_id
          newDataType: bigint
          tableName: host

      # host_aud
       - modifyDataType:
          columnName: id
          newDataType: bigint
          tableName: host_aud
       - modifyDataType:
          columnName: REV
          newDataType: bigint
          tableName: host_aud
       - modifyDataType:
          columnName: REVTYPE
          newDataType: tinyint
          tableName: host_aud
       - modifyDataType:
          columnName: dmp_id
          newDataType: bigint
          tableName: host_aud

      # identifier
       - modifyDataType:
          columnName: id
          newDataType: bigint
          tableName: identifier
       - modifyDataType:
          columnName: version
          newDataType: bigint
          tableName: identifier

      # identifier_aud
       - modifyDataType:
          columnName: id
          newDataType: bigint
          tableName: identifier_aud
       - modifyDataType:
          columnName: REV
          newDataType: bigint
          tableName: identifier_aud
       - modifyDataType:
          columnName: REVTYPE
          newDataType: tinyint
          tableName: identifier_aud

      # inter_storage_translation
       - modifyDataType:
          columnName: id
          newDataType: bigint
          tableName: inter_storage_translation
       - modifyDataType:
          columnName: version
          newDataType: bigint
          tableName: inter_storage_translation
       - modifyDataType:
          columnName: internal_storage_id
          newDataType: bigint
          tableName: inter_storage_translation

      # inter_storage_translation_aud
       - modifyDataType:
          columnName: id
          newDataType: bigint
          tableName: inter_storage_translation_aud
       - modifyDataType:
          columnName: REV
          newDataType: bigint
          tableName: inter_storage_translation_aud
       - modifyDataType:
          columnName: REVTYPE
          newDataType: tinyint
          tableName: inter_storage_translation_aud
       - modifyDataType:
          columnName: internal_storage_id
          newDataType: bigint
          tableName: inter_storage_translation_aud

      # internal_storage
       - modifyDataType:
          columnName: id
          newDataType: bigint
          tableName: internal_storage
       - modifyDataType:
          columnName: version
          newDataType: bigint
          tableName: internal_storage

      # internal_storage_aud
       - modifyDataType:
          columnName: id
          newDataType: bigint
          tableName: internal_storage_aud
       - modifyDataType:
          columnName: REV
          newDataType: bigint
          tableName: internal_storage_aud
       - modifyDataType:
          columnName: REVTYPE
          newDataType: tinyint
          tableName: internal_storage_aud

      # legal_restr_documents
       - modifyDataType:
          columnName: dmp_id
          newDataType: bigint
          tableName: legal_restr_documents

      # legal_restr_documents_aud
       - modifyDataType:
          columnName: dmp_id
          newDataType: bigint
          tableName: legal_restr_documents_aud
       - modifyDataType:
          columnName: REV
          newDataType: bigint
          tableName: legal_restr_documents_aud
       - modifyDataType:
          columnName: REVTYPE
          newDataType: tinyint
          tableName: legal_restr_documents_aud

      # personal_data_compliance
       - modifyDataType:
          columnName: dmp_id
          newDataType: bigint
          tableName: personal_data_compliance

      # personal_data_compliance_aud
       - modifyDataType:
          columnName: dmp_id
          newDataType: bigint
          tableName: personal_data_compliance_aud
       - modifyDataType:
          columnName: REV
          newDataType: bigint
          tableName: personal_data_compliance_aud
       - modifyDataType:
          columnName: REVTYPE
          newDataType: tinyint
          tableName: personal_data_compliance_aud

      # project
       - modifyDataType:
          columnName: id
          newDataType: bigint
          tableName: project
       - modifyDataType:
          columnName: version
          newDataType: bigint
          tableName: project
       - modifyDataType:
          columnName: funding_id
          newDataType: bigint
          tableName: project

      # project_aud
       - modifyDataType:
          columnName: id
          newDataType: bigint
          tableName: project_aud
       - modifyDataType:
          columnName: REV
          newDataType: bigint
          tableName: project_aud
       - modifyDataType:
          columnName: REVTYPE
          newDataType: tinyint
          tableName: project_aud
       - modifyDataType:
          columnName: funding_id
          newDataType: bigint
          tableName: project_aud

      # repository
       - modifyDataType:
          columnName: id
          newDataType: bigint
          tableName: repository

      # repository_aud
       - modifyDataType:
          columnName: id
          newDataType: bigint
          tableName: repository_aud
       - modifyDataType:
          columnName: REV
          newDataType: bigint
          tableName: repository_aud
       - modifyDataType:
          columnName: REVTYPE
          newDataType: tinyint
          tableName: repository_aud

      # revchanges
       - modifyDataType:
          columnName: REV
          newDataType: bigint
          tableName: revchanges

      # revinfo
       - modifyDataType:
          columnName: id
          newDataType: bigint
          tableName: revinfo

      # sensitive_data_security
       - modifyDataType:
          columnName: dmp_id
          newDataType: bigint
          tableName: sensitive_data_security

      # sensitive_data_security_aud
       - modifyDataType:
          columnName: dmp_id
          newDataType: bigint
          tableName: sensitive_data_security_aud
       - modifyDataType:
          columnName: REV
          newDataType: bigint
          tableName: sensitive_data_security_aud
       - modifyDataType:
          columnName: REVTYPE
          newDataType: tinyint
          tableName: sensitive_data_security_aud

      # storage
       - modifyDataType:
          columnName: id
          newDataType: bigint
          tableName: storage
       - modifyDataType:
          columnName: internal_storage_id
          newDataType: bigint
          tableName: storage

      # storage_aud
       - modifyDataType:
          columnName: id
          newDataType: bigint
          tableName: storage_aud
       - modifyDataType:
          columnName: REV
          newDataType: bigint
          tableName: storage_aud
       - modifyDataType:
          columnName: REVTYPE
          newDataType: tinyint
          tableName: storage_aud
       - modifyDataType:
          columnName: internal_storage_id
          newDataType: bigint
          tableName: storage_aud

```
</p>
</details> 

However, this is not possible without disabling/removing foreign key constraints for tables and then enabling/adding these constraints again.
<details><summary>Drop foreign key constraints</summary>


```yaml
  - changeSet:
      id: <some id>
      author: David Eckhard
      changes:
       - dropAllForeignKeyConstraints:
          baseTableName: dataset
       - dropAllForeignKeyConstraints:
          baseTableName: distribution
       - dropAllForeignKeyConstraints:
          baseTableName: host
       - dropAllForeignKeyConstraints:
          baseTableName: personal_data_compliance
       - dropAllForeignKeyConstraints:
          baseTableName: sensitive_data_security
       - dropAllForeignKeyConstraints:
          baseTableName: legal_restr_documents
       - dropAllForeignKeyConstraints:
          baseTableName: dmp_version
       - dropAllForeignKeyConstraints:
          baseTableName: project
       - dropAllForeignKeyConstraints:
          baseTableName: repository
       - dropAllForeignKeyConstraints:
          baseTableName: storage
       - dropAllForeignKeyConstraints:
          baseTableName: storage_aud
       - dropAllForeignKeyConstraints:
          baseTableName: sensitive_data_security_aud

       - addForeignKeyConstraint:
          baseColumnNames: deletion_person_id
          baseTableName: dataset
          constraintName: fk_dataset_contributor
          deferrable:  true
          initiallyDeferred:  true  
          onDelete: SET NULL
          referencedColumnNames: id
          referencedTableName: contributor
          validate: true

  # add additional constraints

```

</details> 

However, this would require some additional manual work to specify all foreign keys again. It will also have to be done again for future database model changes. I tried disabling the checks but to no avail.

These changes are not required per se for a working application, but it would be great to get everything up to date...

#### What does this PR do?
Update quarkus to new version and refactor deprecations.
Add changeset for db update.

No general database structure update is provided (mentioned in the details section). Updating the primary and foreign keys can be done in a separate issue.

#### Breaking changes
ORM update, which requires the db sequence additions. Institutional files may have to be adapted [migration guide](https://github.com/quarkusio/quarkus/wiki/Migration-Guide-3.0:-Hibernate-ORM-5-to-6-migration#one-identifier-generator-sequencetable-per-entity-hierarchy-instead-of-a-single-hibernate_sequence)

#### Code review focus
<!-- What you want the reviewer to focus on -->

#### Dependencies
<!-- If this PR depends on or requires other/additional (frontend) changes, please list them here -->

### Checks
<!-- Adjust list as necessary -->
<!-- In case of DB changes make sure that names do not exceed 30 chars and that audit tables have been created/updated and do not contain FKs on entities. -->
- [ ] Tested with Oracle/PostgreSQL
- [ ] Export updated
- [ ] Documentation added
- [ ] Tests added
- [ ] E2e tests created
- [ ] Successfully ran e2e tests before merge

closes GH-<!-- insert issue number -->

rekt-hard committed:
global: migrate to quarkus version 3.2.0.Final

ValentinFutterer commented on src/main/resources/at/ac/tuwien/damap/db/changeLog-4.x/changeLog-4.0.0_1.yaml:
```diff
@@ -0,0 +1,96 @@
+databaseChangeLog:
```
Is there a reason why the external_storage table did not get a sequence?

ValentinFutterer reviewed (commented):
Looks nice, but exporting the dmp fails - can you look into that

rekt-hard committed:
global: migrate to quarkus v3.11.1

Remove @Lob annotation as it is not required, according to
hibernate documentation: https://docs.jboss.org/hibernate/orm/6.2/introduction/html_single/Hibernate_Introduction.html#column-lengths

rekt-hard commented:
Thank you for noticing this. I updated to the latest version (again) and also removed the `@Lob` annotation, which might have caused this issue. As with a Lob, this would require an explicit transaction within the context of the method. JDBC also expected a long instead of text. Removing this annotation tells the JDBC drive to retrieve the value as text (which it is).

ValentinFutterer reviewed (approved):
Seems like everything works fine noe, nice work!
